### DISCUSSION TITLE
2-part style of LED "neon" printed channel

### DISCUSSION THREAD

**Discussion by ntalekt** [ksjh/LEDneon#4] [Ideas]
I just watched [bigclivedotcom's test "neon" led video](https://www.youtube.com/watch?v=NqlDAagxR5w) and I'd love to figure out how to create a script that would output the 3d-printable led channel using a cool script font. I believe bigclive said the silicone was 8mm and he printed the channel 9mm.

Font examples:
- [Hastron](https://www.dafont.com/hastron.font)
- [Retro Signature](https://www.fontspace.com/retro-signature-font-f53495)
- [Neoncity Script](https://www.fontspace.com/neoncity-script-font-f40955)

Is this already possible using your library? Thanks.

**Comment by ksjh** [comment]
Hi, the fonts you mentioned look great. Unfortunately, it is not so easy to generate LED neon signs the way you would like them automatically. To quote from an earlier (closed) issue:

I was also wondering how we could do this. There are two fundamental problems: Normal fonts just define the outline of glyphs (letters), but we would need the midpoints. I see no obvious way in OpenSCAD to generate these midpoints. The other problem is that the rounding method I used does not work in this case, it needs an area in 2D first.

But I added another module that allows to trace lines. So you need to draw the text you want to trace as lines of neglectable width (e.g. in Inkscape) and use my new module to generate the frame. Please make sure that there are no pointed angles between any lines. Either round them off first (convert them to arcs of suitable size) or make sure that the angles are more or less 90 degrees, so that you can butt two (or three) pieces of LED neon together at the crossing point.
